index.js

```diff
@@ -1,110 +1,90 @@
-// { firstName: 'Tracie', lastName: 'May', balance: '$1,547.73' },
-// { firstName: 'Kendra', lastName: 'Hines', balance: '$12,383.08' },
-// { firstName: 'Kinney', lastName: 'Howard', balance: '$3,207.06' },
-// { firstName: 'Howard', lastName: 'Gilmore', balance: '$21,307.75' },
-// { firstName: 'Rachelle', lastName: 'Schneider', balance: '$35,121.49' },
-// { firstName: 'Lizzie', lastName: 'Alford', balance: '$4,382.94' } ]
-
-// ***************************************************************************
-// Iteration 4 - practice `.filter()` method and how to return two elements
-// ***************************************************************************
+
+// Iteration 4
 
 const genderView = users => {
-  // Your code goes here ...
+  const femaleUsers = users.filter(user => user.gender === 'female').map((user) => {
+    const { firstName, lastName } = user;
+    return {firstName, lastName};
+  });
+  const maleUsers = users.filter(user => user.gender === 'male').map((user) => {
+    const { firstName, lastName } = user;
+    return {firstName, lastName};
+  });
+  return {
+    femaleUsers,
+    maleUsers
+  }
 };
 
 genderView(usersArray);
-// expected output:
-// {
-//    femaleUsers: [ 'Tracie May', 'Kendra Hines', 'Rachelle Schneider', 'Lizzie Alford' ],
-//    maleUsers: [ 'Kirby Doyle', 'Kinney Howard', 'Howard Gilmore' ]
-// }
 
-// ***************************************************************************
 // Bonus - Iteration 5
-// ***************************************************************************
 
 const data = genderView(usersArray);
 
 const genderCount = data => {
-  // Your code goes here ...
+  const {femaleUsers, maleUsers} = data;
+  console.log('Female: ', femaleUsers.length)
+  console.log('Male: ', maleUsers.length)
 };
 
 genderCount(data);
-// expected output:
-// Female: 4
-// Male: 3
 
-// ***************************************************************************
 // Bonus - Iteration 6
-// ***************************************************************************
 
 const promo20 = users => {
-  // Your code goes here ...
+  for (let user of users) {
+    const number = Number(user.balance.substring(1).replace(',', ''));
+  if (number >= 20000) {
+    console.log(`Dear ${user.firstName}, since your balance is ${user.balance}, you are eligible to apply for this awesome credit card.`)
+  }
+}
 };
 
-// expected output:
-// Dear Howard, since your balance is $21,307.75, you are eligible to apply for this awesome credit card.
-// Dear Rachelle, since your balance is $35,121.49, you are eligible to apply for this awesome credit card.
-
-// ***************************************************************************
 // Bonus - Iteration 7
-// ***************************************************************************
 
 const addActive = users => {
-  // Your code goes here ...
+  users.forEach((user) => {
+    user.isActive = true;
+  })
```

Comment:
This function immediately is changing the values you are passing onto it.

In functions, unless said otherwise, try to create a habit that you return a totally new value.
For example
```js
const addActive = users => {
return users.map(user => {
return {
...user, //here you are getting the whole user object copied
isActive: true
}
})
}
```
After this function you get an array of every user active but you get a totally new value.